packages/experimental/Menu/src/MenuTrigger/useMenuTrigger.ts

```diff
@@ -10,15 +11,26 @@ export const useMenuTrigger = (_props: MenuTriggerProps) => {
   const openOnHover = context.openOnHover;
   const triggerRef = context.triggerRef;
 
+  const delayHover = Platform.select({
+    macos: 100,
+    default: 500, // win32
+  });
+
   const onHoverIn = (e: InteractionEvent) => {
     if (openOnHover) {
       setOpen(e, true /* isOpen */);
     }
   };
 
+  const onHoverOut = (e: InteractionEvent) => {
+    if (openOnHover) {
+      setOpen(e, false /* isOpen */);
+    }
+  };
+
   const onClick = (e: InteractionEvent) => {
     setOpen(e, !open);
   };
 
-  return { onClick, onHoverIn, componentRef: triggerRef };
+  return { onClick, onHoverIn, onHoverOut, componentRef: triggerRef, delayHoverIn: delayHover, delayHoverOut: delayHover };
```

Comment:
Just to double check: `delayHoverIn/Out` are forwarded to `useAsPressable`, and this works for submenus too? I remember I had to do a lot of working around that in ContextualMenu

Reply:
I guess submenus is out of scope for this PR. The old ContextualMenu had quite a bit of hardcoding around onHoverIn to present submenus so I mostly wanted to make sure we avoided that. I'll close this comment as out of scope